test/extensions/filters/udp/udp_proxy/session_filters/dynamic_forward_proxy/proxy_filter_test.cc

```diff
@@ -0,0 +1,311 @@
+  setup();
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_()).Times(0);
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, MissingPort) {
+  setup();
+  setFilterStateHost("host");
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_()).Times(0);
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, MissingHost) {
+  setup();
+  setFilterStatePort(50);
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_()).Times(0);
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, EmptyHost) {
+  setup();
+  setFilterState("", 50);
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_()).Times(0);
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, PortIsZero) {
+  setup();
+  setFilterState("", 0);
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_()).Times(0);
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, PortOutOfRange) {
+  setup();
+  setFilterState("", 65536);
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_()).Times(0);
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, RequestOverflow) {
+  setup();
+  setFilterState("host", 50);
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_()).WillOnce(Return(nullptr));
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, CacheLoadOverflow) {
+  setup();
+  setFilterState("host", 50);
+  Upstream::ResourceAutoIncDec* circuit_breakers_{
+      new Upstream::ResourceAutoIncDec(pending_requests_)};
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, canCreateDnsRequest_())
+      .WillOnce(Return(circuit_breakers_));
+  EXPECT_CALL(*dns_cache_manager_->dns_cache_, loadDnsCacheEntry_(Eq("host"), 50, _, _))
+      .WillOnce(Return(
+          MockLoadDnsCacheEntryResult{LoadDnsCacheEntryStatus::Overflow, nullptr, absl::nullopt}));
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onNewSession());
+  EXPECT_EQ(ReadFilterStatus::StopIteration, filter_->onData(recv_data_stub1_));
+}
+
+TEST_F(DynamicProxyFilterTest, AlreadyInCache) {
```

Comment:
Super nit, could you name the tests a bit more descriptively of what's being tested (e.g. `PassesThroughImmediatelyWhenDnsAlreadyInCache` or something like that, in this case; request applies to most test cases)

It makes it easier to see if the tests are checking what they're supposed to, if the name describes what the test is supposed to check.

Reply:
Done, thanks